weave-js/src/core/ops/primitives/date.ts

```diff
@@ -36,6 +36,98 @@ export const opDateSub = makeDateOp({
   },
 });
 
+export const opDateAddDuration = makeDateOp({
+  hidden: true,
+  name: 'date-addDuration',
+  argTypes: {date: 'date', duration: 'number'},
+  renderInfo: {
+    type: 'binary',
+    repr: '+',
+  },
+  description: `Returns the sum between a ${docType(
+    'date'
+  )} and duration in milliseconds ${docType('number')}`,
+  argDescriptions: {
+    date: `Date ${docType('date')}`,
+    duration: `Duration ${docType('number')}`,
+  },
+  returnValueDescription: `The sum between the a ${docType(
+    'date'
+  )} and duration in milliseconds ${docType('number')}`,
+  returnType: inputTypes => 'date',
+  resolver: inputs => {
+    if (
+      inputs.date == null ||
+      inputs.duration == null ||
+      inputs.duration <= 0
+    ) {
+      return null;
+    }
+    const dateAddDuration = moment
+      .utc(inputs.date)
+      .add(moment.duration(inputs.duration, 'milliseconds'))
+      .toDate();
+    return new Date(dateAddDuration);
+  },
+});
+
```

Comment:
In python we have the op

```python
@op(
    name="datetime-add",
    input_type={
        "lhs": types.UnionType(types.Timestamp(), types.TimeDelta()),
        "rhs": lambda input_types: types.optional(
            types.Timestamp()
            if types.TimeDelta().assign_type(input_types["lhs"])
            else types.TimeDelta()
        ),
    },
    output_type=types.optional(types.Timestamp()),
)
def datetime_add(lhs, rhs):
    if rhs == None:
        return None
    return lhs + rhs
```

which is what I think you want to use here. The problem is that JS does not have a notion of a timedelta right now. We should add the timedelta type to JS and implement datetime-add in JS and use it where we would have used this dateAddDuration. 

That way we can use the existing python op